### PR TITLE
Add 5.8 release notes

### PR DESCRIPTION
**Pull request by tomerd**

### CODE REVIEW

File: Documentation/ReleaseNotes/5.8.md [hunk 1]
```diff
@@ -1 +1,21 @@
...
+
+In packages using tools version 5.8 or later, `Foundation` is no longer implicitly imported into package manifests. If `Foundation` APIs are used, the module needs to be imported explicitly.
+  
+## Package Registry Support
+
+SwiftPM now supports token authentication when interacting with a package registry. The `swift package-registry` command has two new subcommands `login` and `logout` as defined in [SE-0378](https://github.com/apple/swift-evolution/blob/main/proposals/0378-package-registry-auth.md) for adding/removing registry credentials.  
+
+## Othe improvments
+
+Improved handling of offline behavior when a cached version of a dependency exists on disk. SwiftPM will check for network availability status to determine if it should attempt to update a checked version of a dependency, and when offline will use the cached version without an update.
```
[the remark this answers is not in the document]
Comment (neonichu) [reply Documentation/ReleaseNotes/5.8.md:17]: I think as written I wouldn't necessarily understand that even a new project/package would be getting whatever is in the shared cache if compatible. Maybe just

> Improved handling of offline behavior when a cached version of a dependency exists on disk in either the per-package or shared cache.

Comment (tomerd) [inline Documentation/ReleaseNotes/5.8.md:17]:
```suggestion
Improved handling of offline behavior when a compatible cached version of a dependency exists on disk in either the per-package or shared cache. SwiftPM will check for network availability status to determine if it should attempt to update a checked version of a dependency, and when offline will use the cached version without an update.  
```

**Review by neonichu — APPROVED** [review]

File: Documentation/ReleaseNotes/5.8.md [hunk 2]
```diff
@@ -1 +1,21 @@
...
+  
+SwiftPM now allows exposing an executable product that consists solely of a binary target that is backed by an artifact bundle. This allow vending binary executables as their own separate package, independently of the plugins that are using them.
+
+In packages using tools version 5.8 or later, `Foundation` is no longer implicitly imported into package manifests. If `Foundation` APIs are used, the module needs to be imported explicitly.
+  
+## Package Registry Support
+
+SwiftPM now supports token authentication when interacting with a package registry. The `swift package-registry` command has two new subcommands `login` and `logout` as defined in [SE-0378](https://github.com/apple/swift-evolution/blob/main/proposals/0378-package-registry-auth.md) for adding/removing registry credentials.  
+
+## Othe improvments
```
Comment (MaxDesiatov) [inline Documentation/ReleaseNotes/5.8.md:15]:
```suggestion
## Other improvments
```

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**